core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -158,20 +161,64 @@ static Snapshot fromJson(FileIO io, JsonNode node) {
     } else {
       // fall back to an embedded manifest list. pass in the manifest's InputFile so length can be
       // loaded lazily, if it is needed
-      List<ManifestFile> manifests =
-          Lists.transform(
-              JsonUtil.getStringList(MANIFESTS, node),
-              location -> new GenericManifestFile(io.newInputFile(location), 0));
```

Comment:
@rdblue do we want to use the `DUMMY_FILE_IO` instance here as well and keep eagerly reading the manifest file locations? Then we could also remove the internal `v1ManifestLocations` variable from `BaseSnapshot`. Let me know what you think.

Reply:
ah nvm, that won't work because we can't read the length with `DUMMY_FILE_IO`